Added Android gradle template for the dataservice-write example (#6)
heremaps/here-olp-edge-sdk-cpp: docs/dataservice-write-example.md

If everything is fine, you would see `Publish Successful - TraceID: <TraceId generated by the platform>`.

> ### Building on Android

Reviewer: should be a h2.
## Building and running on Linux

## Building and running on Android

Author: Done

> Prerequsities: make sure that you have correctly setup the Android environment (downloaded both Android SDK and NDK from official site). Also, make sure that your provided correct keyId, keySecret, and other information stated in the beginning of this README file.

Reviewer: Make Prerequsities a h3
the other two points as unordered list entries.

### Prerequsities
* Setup the Android environment ...
* Provider correct keyId ...

Author: Done

This example shows how to integrate and use the OLP C++ EDGE SDK in Android project.

> First, configure the project with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, path to NDK's Android toolchain file set via `CMAKE_TOOLCHAIN_FILE`, and, optionally, the `ANDROID_` CMake variables, for example to configure SDK version - `ANDROID_PLATFORM`, or ABI for which project will be configured - `ANDROID_ABI`:

Reviewer: ```suggestion
First you need configure the SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, the path to NDK's Android toolchain file set via the `CMAKE_TOOLCHAIN_FILE` variable, and, optionally, the `ANDROID_` CMake variables, e.g. `ANDROID_PLATFORM` or `ANDROID_ABI`.
```

Author: I changed a bit this paragraph considering your changes. Please, take a look again.

```bash
mkdir build && cd build
cmake .. -DEDGE_SDK_BUILD_EXAMPLES=ON -DCMAKE_TOOLCHAIN_FILE=$NDK_ROOT/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a
```

> Executing this `CMake` command will generate a `Gradle` project in the `build/examples/dataservice-write/android` folder. Before using it, you first must install the OLP C++ EDGE SDK libraries into your sysroot:

Reviewer: ```suggestion
`CMake` command will generate a `Gradle` project in the `build/examples/dataservice-write/android` folder. Before it can be used, let's install the OLP C++ EDGE SDK libraries into the sysroot directory:
```

```bash

> (sudo) make install

Reviewer: ```suggestion
# Execute as sudo if necessary
(sudo) make install
```

Author: Done

```

> After running those steps, you can use `gradlew` to build and install the apk on your Android device:

Reviewer: ```suggestion
Now you can use `gradlew` to build and install the apk on your Android device:
```

Author: I changed a bit this paragraph considering your changes. Please, take a look.

```bash
cd build/examples/dataservice-write/android
./gradlew assembleDebug
./gradlew installDebug
```

or you can use the Android Studio IDE by opening the `build.gradle` file located in the `build/examples/dataservice-write/android` folder.

After running the `dataservice_write_example` apk, you should see `Publish Successful` in the main UI screen, if you correctly setup keyId, secretKey, catalog, and layer information. Additional details can be found in the logcat.

> Note, that you can run `CMake` command directly from `examples/dataservice-write/` folder, if you have already installed OLP EDGE SDK libraries. After you did that, make sure that you pass correct dependent libraries paths (OLP EDGE C++ SDK, Snappy, LevelDB) and provided EDGE_NETWORK_PROTOCOL_JAR in the `CMake` command invoked by `build/examples/dataservice-write/android/app/build.gradle` script.

Reviewer: ```suggestion
Note, that you can run `CMake` command directly from `examples/dataservice-write/` folder, if you have already installed OLP EDGE SDK libraries. Make sure that you pass correct library paths (OLP EDGE C++ SDK, Snappy, LevelDB) and provide the correct `EDGE_NETWORK_PROTOCOL_JAR` parameter in the `CMake` command invoked by the `build/examples/dataservice-write/android/app/build.gradle` script.
```

If everything is fine, you would see `Publish Successful - TraceID: <TraceId generated by the platform>`.

## Building and running on Android

> This example shows how to integrate and use the OLP C++ EDGE SDK in Android project.

Reviewer: ```suggestion
This example shows how to integrate and use the OLP C++ EDGE SDK in an Android project.
```

### Prerequisites

> * Setup the Android environment;

Reviewer: ```suggestion
* Setup the Android environment.
```
